ref(notifications): Refactor notification utilities for msteams

97varun opened the pull request:
Move notification utility functions to a common file for reuse in msteams notifications.

* Make a slight modification to the identity query to exclude msteams identity rows which have a `team_id` as `external_id` instead of a `user_id`.

97varun committed:
Remove unused functions

97varun committed:
Replace get_assignee_string

97varun committed:
Use build_attachment_title

97varun committed:
Replace build_group_descr

97varun committed:
Replace build_rule_url

97varun committed:
Build group footer

97varun committed:
Bring back build_group_descr

97varun committed:
Fix incorrect use of format_actor_option

97varun committed:
Use message builder for title and footer

97varun committed:
Fix typing

97varun committed:
Use message builder for assignee note

97varun committed:
Move definition of project

97varun committed:
Use message builder for actions

97varun committed:
test(msteams): Test build group card

97varun committed:
Merge branch '97varun/group-message-builder-tests' into 97varun/card-builder-refactor

97varun committed:
Merge branch '97varun/card-builder-refactor' into 97varun/use-message-builder-basic

97varun committed:
Merge branch '97varun/use-message-builder-basic' into 97varun/card-builder-refactor-actions

97varun committed:
ref(msteams): Create issue message builder class

97varun committed:
Move build_input_choice_card

97varun committed:
Fix package resolution issue

97varun committed:
Merge branch '97varun/use-message-builder-basic' into 97varun/card-builder-refactor-actions

97varun committed:
Move issue builder class to separate file

97varun committed:
Remove unnecessary setup

97varun committed:
Merge branch 'master' into 97varun/group-message-builder-tests

97varun committed:
Merge branch '97varun/group-message-builder-tests' into 97varun/card-builder-refactor

97varun committed:
Merge branch '97varun/card-builder-refactor' into 97varun/use-message-builder-basic

97varun committed:
Merge branch '97varun/use-message-builder-basic' into 97varun/card-builder-refactor-actions

97varun committed:
Merge branch '97varun/card-builder-refactor-actions' into 97varun/issue-message-builder-class

97varun committed:
Add type hints

97varun committed:
Fix image block and typing

97varun committed:
Merge branch '97varun/use-message-builder-basic' into 97varun/card-builder-refactor-actions

97varun committed:
Merge branch '97varun/card-builder-refactor-actions' into 97varun/issue-message-builder-class

97varun committed:
Add note for text to block conversion

97varun committed:
Merge branch '97varun/card-builder-refactor-actions' into 97varun/issue-message-builder-class

97varun committed:
Move common function to separate file

97varun committed:
Add url_format to build_footer

97varun committed:
Merge branch 'master' into 97varun/group-message-builder-tests

97varun committed:
Merge branch '97varun/group-message-builder-tests' into 97varun/card-builder-refactor

97varun committed:
Merge branch '97varun/card-builder-refactor' into 97varun/use-message-builder-basic

97varun committed:
Merge branch '97varun/use-message-builder-basic' into 97varun/card-builder-refactor-actions

97varun committed:
Merge branch '97varun/card-builder-refactor-actions' into 97varun/issue-message-builder-class

97varun committed:
Merge branch '97varun/issue-message-builder-class' into 97varun/common-util-functions

97varun committed:
Rename notification -> message_builder

97varun committed:
Remove unnecessary setup

97varun committed:
Merge branch '97varun/group-message-builder-tests' into 97varun/card-builder-refactor

97varun committed:
Merge branch '97varun/card-builder-refactor' into 97varun/use-message-builder-basic

97varun committed:
Merge branch '97varun/use-message-builder-basic' into 97varun/card-builder-refactor-actions

97varun committed:
Merge branch '97varun/card-builder-refactor-actions' into 97varun/issue-message-builder-class

97varun committed:
Merge branch '97varun/issue-message-builder-class' into 97varun/common-util-functions

97varun committed:
ref(notifications): Refactor notification utilities for msteams

97varun commented on src/sentry/integrations/notifications.py:
```diff
@@ -0,0 +1,98 @@
+        # For Microsoft Teams integration, initially we create rows in the
+        # identity table with the external_id as a team_id instead of the user_id.
+        # We need to exclude rows where this is NOT updated to the user_id later.
+        .exclude(external_id=F("idp__external_id")).select_related("idp")
```
Rows where `identity.external_id == idp.external_id` will be excluded.

97varun committed:
Merge branch 'master' into 97varun/card-builder-refactor-actions

97varun committed:
Merge branch 'master' into 97varun/card-builder-refactor-actions

97varun committed:
Undo inadvertent change during merge

97varun committed:
Merge branch 'master' into 97varun/card-builder-refactor-actions

97varun committed:
Merge branch '97varun/card-builder-refactor-actions' into 97varun/issue-message-builder-class

97varun committed:
Merge branch '97varun/issue-message-builder-class' into 97varun/common-util-functions

97varun committed:
Fix mypy typing error

97varun committed:
Fix typing issues

97varun committed:
Merge branch '97varun/issue-message-builder-class' into 97varun/common-util-functions

getsantry[bot] committed:
style(lint): Auto commit lint changes

97varun committed:
Update call to get_integrations_by_channel_by_recipient

97varun committed:
Merge branch 'master' into 97varun/common-util-functions

97varun committed:
Merge branch 'master' into 97varun/common-util-functions

97varun committed:
Merge branch 'master' into 97varun/common-util-functions

AniketDas-Tekky commented on src/sentry/integrations/notifications.py:
```diff
@@ -0,0 +1,98 @@
+    return {external_actor.external_id: external_actor.integration}
+
+
+def get_integrations_by_channel_by_recipient(
```
Are the tests updated to call this instead?

97varun replied:
@AniketDas-Tekky, thanks for pointing out! This function wasn't being tested directly. I added some tests which tests the new functionality as well.

97varun committed:
Add tests

AniketDas-Tekky commented on tests/sentry/integrations/test_notification_utilities.py:
```diff
@@ -0,0 +1,68 @@
+
+        self.integration = install_slack(self.organization)
+
+        self.idp = IdentityProvider.objects.create(type="slack", external_id="TXXXXXXX1", config={})
```
we have fixtures to help with setup. In this case `self.create_identity_provider`

https://github.com/getsentry/sentry/blob/master/src/sentry/testutils/fixtures.py#L391

You can do the same with identity

97varun replied:
This is so good, reduced a lot of boilerplate.

AniketDas-Tekky commented on tests/sentry/integrations/test_notification_utilities.py:
```diff
@@ -0,0 +1,68 @@
+
+        self.idp = IdentityProvider.objects.create(type="slack", external_id="TXXXXXXX1", config={})
+        self.identity = Identity.objects.create(
+            external_id="UXXXXXXX1",
```
nit: save strings as test constants
```
self.external_id_1 = "UXXXXXXX1"
# ...
    external_id=self.external_id_1
```

97varun replied:
Updated.

97varun committed:
Use fixtures

97varun committed:
Merge branch 'master' into 97varun/common-util-functions

97varun committed:
Merge branch '97varun/common-util-functions' into 97varun/refactor-common-notif-functions

lynnagara committed:
ref: Remove unused setting (#36971)

This was used temporarily to roll out some changes and is no longer
referenced in the codebase.

jesus4497 committed:
feat(replays): network table decent basic styles (#36949)

Added new styles for the network table and added a new sort functionality.

NOTE: 
We still don't have statuses values in our network span object so I just put a placeholder in the meantime!

Closes #36587

jesus4497 committed:
fix(replays): header styles and project badge icon (#36973)

Fix styles and project badge icon using the hook `useProjects()` in order to get the correct data for `<ProjectBadge>`. 

Closes #36771 
Closes #36772
Closes #36110

priscilawebdev committed:
ref(device-context): Replace getMeta (proxy) with _meta object - (#36966)

priscilawebdev committed:
ref(runtime-context): Replace getMeta (proxy) with _meta object - (#37006)

priscilawebdev committed:
ref(trace-context): Replace getMeta (proxy) with _meta object -  (#37010)

priscilawebdev committed:
ref(gpu-context): Replace getMeta (proxy) with _meta object - (#37013)

priscilawebdev committed:
ref(app-context): Replace getMeta (proxy) with _meta object -  (#37042)

priscilawebdev committed:
ref(os-context): Replace getMeta (proxy) with _meta object - (#37040)

wmak committed:
fix(mep): Fallback threshold functions (#37025)

- This updates the metric config so that apdex, count_miserable and
  user-misery will raise IncompatibleMetricsQuery. That way when the
  dataset is MetricsEnhanced the query will fallback to the discover
  dataset

priscilawebdev committed:
ref(context-summary-context): Replace getMeta (proxy) with _meta object (#37046)

wmak committed:
feat(custom-measurements): Add units to events stats (#36977)

- This splits events stats tests apart to a separate mep file
- This adds a meta key to the stats response with units/fields to match
  events

priscilawebdev committed:
ref(browser-context): Replace getMeta (proxy) with _meta object - (#37041)

edwardgou-sentry committed:
feat(discover): Show Custom Performance Metrics on Event Detail pages (#37011)

Displays Custom Performance Metrics on the Event Detail page and Performance Event Detail Page if they exist

priscilawebdev committed:
ref(sampling): Attempt to fix e2e flaky test (#36790)

lynnagara committed:
feat(post-process-forwarder): Support passing topic as CLI arg (#36983)

We are going to split the transactions topic from the events topic so errors
and transactions have independent infrastructure. We will do a gradual transition
where a subset of projects will be migrated over first. In order to support this
we will have to temporarily run two post process forwarders (one for transactions on
the old topic and one for the new one). For this we need to be able to pass the topic to
consume from via CLI rather than just using the default one.

armcknight committed:
fix: update ios device classification (#37044)

This classification was written some time ago and is falling out of date. Two main reasons for this to be revised:

1. There are now devices clocking over 3 GHz.
2. Core count is no longer a useful metric as the only device with a 4-core CPU–the iPhone 8–only uses two at a time (either both performance cores or both efficiency cores). The next model has 6 cores, but only 4 that perform at the max clock speed, with 2 efficiency cores at much lower rates. Just as well, the type of work cores do have diversified into motion and image coprocessors and neural engines (in addition to the GPU, of course).

For now we'll simply use the following classification based on CPU clock rate, since we are still primarily focusing on CPU usage for profiling purposes:

low end: < 2 GHz (iPhone 4s to iPhone 6s)
mid: ≥ 2 and < 3 GHz (iPhone 7 to iPhone 11)
high: ≥ 3 GHz (iPhone 12 and later)

This roughly partitions the current history of devices we support into thirds.

narsaynorath committed:
feat(dashboard-filters): Allow users to save (#36944)

Users can save dashboard filters. If there are unsaved filters and
the user does Edit Dashboard, or any of the widget update actions,
we wipe the unsaved filters for now.

leeandher committed:
fix(ui): Change behaviour of AsyncSelect field for SentryApps (#37002)

This PR addresses a few bugs with the Async Select fields used by Sentry Apps

lynnagara committed:
build: Bump arroyo to 1.0.0 (#37033)

k-fish committed:
feat(mep): Modify sidebar charts on summary (#37051)

* feat(mep): Modify sidebar charts on summary

This modifies the Sidebar charts on the transaction summary to show both transaction and metric data.

joshuarli committed:
feat(ci): dispatch bump-sentry (#35981)

97varun committed:
Merge branch 'master' into 97varun/refactor-common-notif-functions
